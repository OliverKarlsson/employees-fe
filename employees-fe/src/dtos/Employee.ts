interface Employee {
  name: string;
  email: string;
  phoneNumber: string | null;
  office: string | null;
  manager: string | null;
  orgUnit: string;
  // string of HTML
  mainText: string | null;
  gitHub: string | null;
  twitter: string | null;
  stackOverflow: string | null;
  linkedIn: string | null;
  imagePortraitUrl: string | null;
  imageWallOfLeetUrl: string | null;
  highlighted: boolean;
  published: boolean;
  primaryRole: string | null;
  secondaryRole: string | null;
  area: string | null;
}

export default Employee;
