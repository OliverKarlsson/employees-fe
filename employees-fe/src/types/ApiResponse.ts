interface ApiResponse<T> {
  status: ApiResponseStatus;
  responseData: T | null;
  error: Error | null;
}

export default ApiResponse;
