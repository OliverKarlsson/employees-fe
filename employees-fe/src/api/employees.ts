import Employee from "dtos/Employee";
import isEmployees from "homeless/isEmployees";
import ApiCall from "types/ApiCall";
import { baseUrl, apiKey } from "config";

/**
 * Fetches all employees
 * @returns
 */
export const getAllEmployees: ApiCall<Employee[]> = async () => {
  try {
    const response = await fetch(baseUrl, {
      headers: {
        Authorization: apiKey,
      },
    });
    const responseData = await response.json();
    if (!isEmployees(responseData)) {
      throw new Error(
        "Received unsuported employees in the `getAllEmployees` call. The response data was expected to be of type `Employee[]` etc..."
      );
    }
    return { status: "loaded", responseData, error: null };
  } catch (error) {
    return { status: "failed", responseData: null, error: error as Error };
  }
};
