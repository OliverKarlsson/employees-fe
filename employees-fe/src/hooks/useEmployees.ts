import { getAllEmployees } from "api/employees";
import Employee from "dtos/Employee";
import { useEffect, useMemo, useState } from "react";
import ApiResponse from "types/ApiResponse";
import EmployeesSort from "types/EmployeesSort";
import EmployeesView from "types/EmployeesView";

const highString = "������";

type EmployeeStringProperty = keyof Omit<Employee, "highlighted" | "published">;

const makeEmployeeStringCompare =
  (propertyName: EmployeeStringProperty) => (a: Employee, b: Employee) =>
    (a[propertyName] ?? highString).localeCompare(
      b[propertyName] ?? highString
    );

const useEmployees = () => {
  const [employeesResponse, setEmployeesResponse] = useState<
    ApiResponse<Employee[]>
  >({ status: "initial", responseData: null, error: null });

  const [activeFilter, setActiveFilter] = useState("");
  const [activeView, setActiveView] = useState<EmployeesView>("grid");
  const [activeSort, setActiveSort] = useState<EmployeesSort>("name");

  useEffect(() => {
    let isCurrent = true;
    getAllEmployees().then((response) => {
      if (isCurrent) {
        setEmployeesResponse(response);
      }
    });
    return () => {
      isCurrent = false;
    };
  }, [setEmployeesResponse]);

  const employees = useMemo<Employee[]>(() => {
    if (!employeesResponse.responseData) {
      return [];
    }
    const filteredEmployees = employeesResponse.responseData.filter(
      ({ name, office }: Employee) => {
        const lowerCaseFilter = activeFilter.toLowerCase();
        return (
          name.toLowerCase().includes(lowerCaseFilter) ||
          office?.toLowerCase().includes(lowerCaseFilter)
        );
      }
    );

    const sortedEmployees = filteredEmployees.sort(
      makeEmployeeStringCompare(activeSort)
    );

    return sortedEmployees;
  }, [employeesResponse, activeFilter, activeSort]);

  return {
    employees,
    activeFilter,
    activeView,
    activeSort,
    onUpdateSort: setActiveSort,
    onUpdateFilter: setActiveFilter,
    onUpdateView: setActiveView,
  };
};

export default useEmployees;
