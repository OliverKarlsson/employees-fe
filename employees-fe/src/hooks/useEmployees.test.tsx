import { expect, it, describe, vi, afterEach } from "vitest";
import useEmployees from "./useEmployees";
import Employee from "dtos/Employee";
import employees from "__fixtures__/employees";
import { fireEvent, render, waitFor } from "@testing-library/react";
import { getAllEmployees } from "api/employees";
import { useEffect } from "react";
import ApiResponse from "types/ApiResponse";

const p = new Promise<ApiResponse<Employee[]>>((resolve) => {
  resolve({ status: "loaded", responseData: employees, error: null });
});

vi.mock("../api/employees", () => ({
  getAllEmployees: vi.fn().mockImplementation(() => p),
}));
describe("useEmployees", () => {
  it("should be a function", () => {
    expect(useEmployees).toBeInstanceOf(Function);
  });
  describe("when employees are fetched successfully", () => {
    afterEach(() => {
      (getAllEmployees as unknown as any).mockClear();
    });

    it("should fetch employees only once", () => {
      const TestComponent = () => {
        useEmployees();
        return null;
      };
      render(<TestComponent />);
      expect(getAllEmployees).toHaveBeenCalledOnce();
    });

    it("should have correct default values (including sort)", async () => {
      const TestComponent = () => {
        const { activeSort, activeFilter, activeView, employees } =
          useEmployees();
        return (
          <>
            <script id="data" type="application/json">
              {JSON.stringify({
                activeFilter,
                activeView,
                activeSort,
                employees: employees.slice(0, 10),
              })}
            </script>
          </>
        );
      };
      const { container } = render(<TestComponent />);
      expect(container.firstChild?.textContent).toMatchSnapshot();
      await waitFor(() =>
        expect(
          JSON.parse(container.firstChild?.textContent ?? "").employees
        ).not.toHaveLength(0)
      );
      expect(container.firstChild?.textContent).toMatchSnapshot();
    });

    it("should update sort successfully", async () => {
      const TestComponent = () => {
        const { activeSort, employees, onUpdateSort } = useEmployees();
        return (
          <>
            <script id="data" type="application/json">
              {JSON.stringify({
                activeSort,
                employees: employees.slice(0, 50),
              })}
            </script>
            <button onClick={() => onUpdateSort("office")} />
          </>
        );
      };
      const { container } = render(<TestComponent />);
      await waitFor(() =>
        expect(
          JSON.parse(container.firstChild?.textContent ?? "").employees
        ).not.toHaveLength(0)
      );
      container.lastChild && fireEvent.click(container.lastChild);

      expect(
        JSON.parse(container.firstChild?.textContent ?? "").employees.map(
          ({ name, office }: any) => name + ":" + office
        )
      ).toMatchSnapshot();
    });

    it("should update filter successfully", async () => {
      const TestComponent = () => {
        const { activeSort, employees, onUpdateFilter } = useEmployees();
        return (
          <>
            <script id="data" type="application/json">
              {JSON.stringify({
                activeSort,
                employees: employees.slice(0, 50),
              })}
            </script>
            <button onClick={() => onUpdateFilter("ING")} />
          </>
        );
      };
      const { container } = render(<TestComponent />);
      await waitFor(() =>
        expect(
          JSON.parse(container.firstChild?.textContent ?? "").employees
        ).not.toHaveLength(0)
      );
      container.lastChild && fireEvent.click(container.lastChild);

      expect(
        JSON.parse(container.firstChild?.textContent ?? "").employees.map(
          ({ name, office }: any) => name + ":" + office
        )
      ).toMatchSnapshot();
    });

    it("should remove filter successfully", async () => {
      const TestComponent = () => {
        const { activeSort, employees, onUpdateFilter } = useEmployees();
        useEffect(() => {
          onUpdateFilter("ING");
        }, [onUpdateFilter]);
        return (
          <>
            <script id="data" type="application/json">
              {JSON.stringify({
                activeSort,
                employees: employees.slice(0, 50),
              })}
            </script>
            <button onClick={() => onUpdateFilter("")} />
          </>
        );
      };
      const { container } = render(<TestComponent />);
      await waitFor(() =>
        expect(
          JSON.parse(container.firstChild?.textContent ?? "").employees
        ).not.toHaveLength(0)
      );
      container.lastChild && fireEvent.click(container.lastChild);

      expect(
        JSON.parse(container.firstChild?.textContent ?? "").employees.map(
          ({ name, office }: any) => name + ":" + office
        )
      ).toMatchSnapshot();
    });
  });

  describe.todo("when employees are fetched unsuccessfully");
});
