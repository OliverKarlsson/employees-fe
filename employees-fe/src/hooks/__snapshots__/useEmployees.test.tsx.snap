// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useEmployees > when employees are fetched successfully > should have correct default values (including sort) 1`] = `"{\\"activeFilter\\":\\"\\",\\"activeView\\":\\"grid\\",\\"activeSort\\":\\"name\\",\\"employees\\":[]}"`;

exports[`useEmployees > when employees are fetched successfully > should have correct default values (including sort) 2`] = `"{\\"activeFilter\\":\\"\\",\\"activeView\\":\\"grid\\",\\"activeSort\\":\\"name\\",\\"employees\\":[{\\"name\\":\\"Abdallah Safarini\\",\\"email\\":\\"abdallah.safarini@1337.tech\\",\\"phoneNumber\\":\\"+46725133776\\",\\"office\\":\\"Lund\\",\\"manager\\":\\"andreas.nilsson@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>Seeking more challenges lead me to 13/37; Curiosity, Intellectual conversations, and Challenges drive me the most and I go with their flow usually but as much as it sounds rebellious, I tend to control/use them to shape the output in a calm steady personality to be progressive in my career mainly and also in my personality development and my adaptation skills</p><p>Out of work, I enjoy different types of music and l like/enjoy watching good actors in movies and series. Besides that, I like walking, swimming and going to GYM classes from time to time</p>  \\",\\"gitHub\\":null,\\"twitter\\":null,\\"stackOverflow\\":null,\\"linkedIn\\":\\"/in/abdullah-s-b3b231a2/\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/abdallah-safarini\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/abdallah-safarini\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":\\"Backend engineer\\",\\"secondaryRole\\":null,\\"area\\":\\"Engineering\\"},{\\"name\\":\\"Agron Kabashi\\",\\"email\\":\\"agron.kabashi@1337.tech\\",\\"phoneNumber\\":\\"+46725133736\\",\\"office\\":\\"Lund\\",\\"manager\\":\\"lotta.wennolf@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>I have been working in the software industry for over 10 years now, and even though the road has been bumpy, I don’t regret a single second of it! My experiences have been quite varied, from creating games to full blown web-based intranet platforms.</p><p>Because of this variation and these experiences I’ve found that I’ve acquired a thirst for front-end development and anything web related, which can range from mobile apps to web-based SaaS solutions. The fact that you can create ANYTHING with almost nothing is an exhilarating feeling. It’s one of the major reasons that keeps me going; the only limit is my imagination.</p><p>I absolutely love to travel and experience new cultures (and food!) when the opportunity presents itself. My goal in life is to visit as many places and experience as many different cultures as I possibly can. So far I can cross off Europe and half of Asia from my list, and I look forward to many more exciting adventures to come.</p>   \\",\\"gitHub\\":\\"AgronKabashi\\",\\"twitter\\":\\"_agronkabashi\\",\\"stackOverflow\\":null,\\"linkedIn\\":\\"/pub/agron-kabashi/54/6a8/4a6\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/agron-kabashi\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/agron-kabashi\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":null,\\"secondaryRole\\":null,\\"area\\":\\"Engineering\\"},{\\"name\\":\\"Ahmad Mirzaei\\",\\"email\\":\\"ahmad.mirzaei@1337.tech\\",\\"phoneNumber\\":\\"+46705133787\\",\\"office\\":\\"Lund\\",\\"manager\\":\\"sarah.thoren@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>It all started with my brother's projects from local businesses almost 8 years ago, when he was responsible for developing web interfaces for them.</p> <p>There was a time where he was struggling to handle multiple projects at the same time, so he encouraged me to join him in the journey. After completing some small projects - using simple UI designs - it became my passion.This passion drives me even up until this moment and it still makes me forget about the time as I focus on providing a better solution for the challenges that I’m facing all the time. Yes, this passion that I am talking about is like an engine inside me that would work without fuel, something infinite, only for one reason and that is I love what I do.</p> <p> I have always believed that Front-end development is a thing that I would love, because it always brings me some fresh challenges to think about, makes me use all my effort and thought in order to provide a better solution.Thinking about the years that have passed by so far - deep diving in heart of the front end technologies and finding out about all it’s nuts and bolts and filling out my days and nights with it - brings a smile to my face. This smile is because I can look at myself and know that I've been enjoying some of the best times of my life, and I’m really glad that I've chosen the right thing for rest of it.</p> <p>Tech life aside, in my spare time I always love to go biking, hiking, and love to walk or run. I’m not good at cooking but I actually love the experience of cooking when there are no options left (only for survival purpose). If I’m not doing one of those things, you would probably find me in Github’s explore page or in stack overflow helping some tech fellows.</p>   \\",\\"gitHub\\":\\"a-m-dev\\",\\"twitter\\":\\"a_m_dev\\",\\"stackOverflow\\":\\"5354341\\",\\"linkedIn\\":\\"/in/ahmad-mirzaei-b60b2618a/\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/ahmad-mirzaei\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/ahmad-mirzaei\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":null,\\"secondaryRole\\":null,\\"area\\":\\"Engineering\\"},{\\"name\\":\\"Ahmed Bazzara\\",\\"email\\":\\"ahmed.bazzara@1337.tech\\",\\"phoneNumber\\":\\"+46705133723\\",\\"office\\":\\"Lund\\",\\"manager\\":\\"andreas.nilsson@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>It started with a hedgehog, not just any hedgehog. Sonic The Hedgehog was 9 years old, I was five. Since then gaming and computers has been the most persistent distraction of my life. Despite this I’m a very social and outgoing creature who values the company of my friends highly.</p><p>It took me a while to find programming -were around 11 years old when I wrote my first program on VB-, but when I finally did, I realised that I’m totally hooked on the thought of creating, finding flaws and develop smart digital solutions. I keep chasing new knowledge in order to improve and I’m one of those who get truly satisfied when I can see the results of my work. </p><p>The last couple of years I’ve been struggling to find my way as a newbie in Sweden. Now when I’ve finally found it I feel ready to dive in to the world of programming for real. I like the thought diving and I’m actually also a quite dedicated swimmer in my spare time. Every now and then I love to marathon watch a really good TV-series and If I’m not inside you’ll find me camping in the nature. I haven’t got to do that much travelling the last couple of years, but the very second I get the opportunity I’ll be on an airplane ready to explore and fill my self with experiences.</p>   \\",\\"gitHub\\":\\"ticooo95\\",\\"twitter\\":null,\\"stackOverflow\\":null,\\"linkedIn\\":\\"/in/ahmed-bazzara/\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/ahmed-bazzara\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/ahmed-bazzara\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":null,\\"secondaryRole\\":null,\\"area\\":\\"Engineering\\"},{\\"name\\":\\"Akan Murat Cimen\\",\\"email\\":\\"akan.murat.cimen@1337.tech\\",\\"phoneNumber\\":\\"+46761337198\\",\\"office\\":\\"Lund\\",\\"manager\\":\\"jim@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>It has always been appealing to me to try to understand everything from the big bang to AI-suggested cat videos by asking the what, why, and how questions. If I ask the right questions at the right time, in the right order, I believe everything has an answer. The knowledge we have learned as human beings so far by asking these questions is an invaluable treasure.</p><p>I have been demanding some operations from CPUs since I was a kid. I worked on various projects like e-commerce, content management systems, car sharing, live streaming etc. with various programming languages. I think the true \\\\\\"legacy code\\\\\\" is not about how old it is, it's about its readability, maintainability etc. I'm trying not to be that person who writes code like that.</p><p>In my spare time I'm demanding more operations from CPUs by playing games, watching movies, tv-series or traveling with the help of a navigation application.</p> \\",\\"gitHub\\":\\"akanmuratcimen\\",\\"twitter\\":null,\\"stackOverflow\\":\\"513234\\",\\"linkedIn\\":\\"/in/akanmuratcimen/\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/akan-murat-cimen\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/akan-murat-cimen\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":null,\\"secondaryRole\\":null,\\"area\\":null},{\\"name\\":\\"Alban Nurkollari\\",\\"email\\":\\"alban.nurkollari@1337.tech\\",\\"phoneNumber\\":\\"+46768133799\\",\\"office\\":\\"Helsingborg\\",\\"manager\\":\\"andreas.nilsson@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>My friends and colleagues describe me as a friendly, helping, adapting, energetic and most of all focus-oriented person.</p><p>I embrace programming with creativity and try to provide best practices and logic. I value diversity among peers and respect them utmost.</p>   \\",\\"gitHub\\":\\"albannurkollari\\",\\"twitter\\":\\"VennBani\\",\\"stackOverflow\\":\\"5453170\\",\\"linkedIn\\":\\"/in/albannurkollari/\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/alban-nurkollari\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/alban-nurkollari\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":null,\\"secondaryRole\\":null,\\"area\\":\\"Engineering\\"},{\\"name\\":\\"Alexander Baldwin\\",\\"email\\":\\"alexander.baldwin@1337.tech\\",\\"phoneNumber\\":\\"+46706161337\\",\\"office\\":\\"Lund\\",\\"manager\\":\\"sarah.thoren@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>After studying for a degree in mathematics in the United Kingdom (my homeland), I found my way to Sweden where I realized that I really wanted to write code for a living. Recalling a short-but-rewarding stint doing development for a Neverwinter Nights persistent world back in my teenage years (and maybe also a bit too much time spent playing Starcraft), I decided to study video games development. Since then, I've mainly been doing web development in the simulation industry - creating powerful ways for users to solve real engineering problems in a web browser!</p><p>Away from the computer, I enjoy growing my own vegetables, fermenting them, and then - of course - cooking with them! I'm also an avid reader, love music, and try to still fit in time for some gaming when I get a chance.</p>  \\",\\"gitHub\\":null,\\"twitter\\":null,\\"stackOverflow\\":null,\\"linkedIn\\":\\"/in/alexander-baldwin-b0a419170/\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/alexander-baldwin\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/alexander-baldwin\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":null,\\"secondaryRole\\":null,\\"area\\":\\"Engineering\\"},{\\"name\\":\\"Alexander Danson\\",\\"email\\":\\"alexander.danson@1337.tech\\",\\"phoneNumber\\":\\"+46727133766\\",\\"office\\":\\"Stockholm\\",\\"manager\\":\\"johan.lundborg@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>I'm originally from the heart of Lancashire, in Northern England before moving to Sweden in 2014 after meeting my wife Caroline. I really enjoy the feeling of creating something new with the help of the latest technology, with a healthy portion of team spirit. What I love about being a project manager and scrum master is the mutual sense of accomplishment when the team succeeds, and also the opportunity to overcome and learn from a variety of challenges along the way. I also enjoy coding where possible, exploring new technologies to provide me with inspiration and insights which could be applied in my work.</p><p>When I'm not working, I love playing the guitar (electric or acoustic depending on mood!), so let me know if you want to jam! I also enjoy keeping fit and have a new-found interest in cross-country skiing after moving to Sweden, and have since completed En Svensk Klassiker and 2 Vasalopp. Let's see what the next challenge will be!</p><p>As an Englishman I obviously enjoy a day of watching Premier League (Liverpool is my team), accompanied by a slightly warm ale. I also like good dose of British Comedy, such as The Office (UK), Alan Partridge, Hot Fuzz or Stewart Lee to name a few. I might drop in a few random quotes in conversation, so don't be surprised if I start laughing inexplicably at something I've just said.</p>        \\",\\"gitHub\\":\\"alexanderdanson\\",\\"twitter\\":\\"alexsdanson\\",\\"stackOverflow\\":null,\\"linkedIn\\":\\"/in/alexdanson/\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/alexander-danson\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/alexander-danson\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":null,\\"secondaryRole\\":null,\\"area\\":\\"Product Management\\"},{\\"name\\":\\"Alicia Sjöberg\\",\\"email\\":\\"alicia.sjoberg@1337.tech\\",\\"phoneNumber\\":\\"+46702326739\\",\\"office\\":\\"Stockholm\\",\\"manager\\":\\"tommy.albinsson@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>I am a caring person who is interested in understanding the people I meet. I believe in being open, true, and that great communication is the key to almost all complex situations between people.</p><p>I believe working with Talent Acquisition is the perfect combo of getting-to-know and to help different types of people. I thrive in a fast-paced environment where I get to face different kinds of challenges frequently.</p><p>Nothing is more challenging and at the same time, much awarding as working with people. Working within the tech area makes me feel like I am contributing to the future, which is an important factor for me.</p><p>When I'm not working, I like to stay as active as possible without disturbing the life balance. I like to enjoy all the fun and beautiful things in life: family and friends, food and beverages, nature, physical activities, animals etc. (It's a long list).</p>  \\",\\"gitHub\\":null,\\"twitter\\":null,\\"stackOverflow\\":null,\\"linkedIn\\":\\"/in/alicia-sjöberg-42715014b/\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/alicia-sjoberg\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/alicia-sjoberg\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":null,\\"secondaryRole\\":null,\\"area\\":null},{\\"name\\":\\"Alireza Rafeizadeh\\",\\"email\\":\\"alireza.rafeizadeh@1337.tech\\",\\"phoneNumber\\":\\"+46761337387\\",\\"office\\":\\"Lund\\",\\"manager\\":\\"lotta.wennolf@1337.tech\\",\\"orgUnit\\":\\"/Employees\\",\\"mainText\\":\\"<p>I have studied mathematics at university. Math and software development have a lot in common and at the core, they involve a lot of problem solving through logical deduction.I believe software makes people’s lives easier and working to achieve this goal is what describes me. I write software for android devices. System design and UI/UX are two other areas of my interest and I try to get myself updated bout them.</p> <p>I got introduced to computers when I was 11 and after getting enough of playing some games, I started to learn 3ds Max because animations inspired me a lot. I had no idea of why exactly I am learning this or if it will come in handy in the future. It was just something I enjoyed doing. Years later I figured out that software production is what I really enjoy, so I chose it as my career.</p> <p>In my free time, I read and watch a lot of gadget related stuff and consider myself as a tech-savvy! I have also a great passion for art. I have some experience with calligraphy and in-lay-on wood. However, music is something else! I play around with FL studio and Logic pro as a not-professional-but-very-interested music fan!</p>   \\",\\"gitHub\\":\\"alireza-rafeezadeh\\",\\"twitter\\":\\"@novus_arcana\\",\\"stackOverflow\\":null,\\"linkedIn\\":\\"/in/alirezarafee/\\",\\"imagePortraitUrl\\":\\"https://i.1337co.de/profile/alireza-rafeizadeh\\",\\"imageWallOfLeetUrl\\":\\"https://i.1337co.de/wallofleet/alireza-rafeizadeh\\",\\"highlighted\\":false,\\"published\\":true,\\"primaryRole\\":null,\\"secondaryRole\\":null,\\"area\\":\\"Engineering\\"}]}"`;

exports[`useEmployees > when employees are fetched successfully > should remove filter successfully 1`] = `
[
  "Abdallah Safarini:Lund",
  "Agron Kabashi:Lund",
  "Ahmad Mirzaei:Lund",
  "Ahmed Bazzara:Lund",
  "Akan Murat Cimen:Lund",
  "Alban Nurkollari:Helsingborg",
  "Alexander Baldwin:Lund",
  "Alexander Danson:Stockholm",
  "Alicia Sjöberg:Stockholm",
  "Alireza Rafeizadeh:Lund",
  "Amin Yosoh:Lund",
  "Amirmasoud Ramezani:Stockholm",
  "Anastasiia Petrych:Stockholm",
  "Anatolii Avramenko:Lund",
  "Anders Clark:Lund",
  "Anders Hedlund:Borlänge",
  "Anders Iggsten:Stockholm",
  "Anders Liljekvist:Borlänge",
  "Anders Ringqvist:Lund",
  "André Brdoch:Stockholm",
  "Andreas Cederström:Lund",
  "Andreas Håkansson:Helsingborg",
  "Andreas Nilsson:Helsingborg",
  "Andreas Olsson:Lund",
  "Andreas Voigt:Helsingborg",
  "Ani Subotić:Ljubljana",
  "Anna Van Duijvenbode:Lund",
  "Anneli Idnert:Lund",
  "Anton Maskaikin:Stockholm",
  "Anton Panteleev:Stockholm",
  "Anže Stražar:Ljubljana",
  "Ariton Karamani:Lund",
  "Arman Momeni:Lund",
  "Artem Syromiatnikov:Stockholm",
  "Axel Olsson:Lund",
  "Bastian Dehlén:Stockholm",
  "Biserka Cvetkovska:Ljubljana",
  "Blake Lundquist:Stockholm",
  "Britta Patterson:Stockholm",
  "Carl Glennfalk:Stockholm",
  "Carl Nilsson:Lund",
  "Carl Stenberg:Lund",
  "Cedrik Aagaard:Lund",
  "Christian Söderberg:Stockholm",
  "Christian Tallhamn:Lund",
  "Christof Lindmark:Lund",
  "Christoffer Hydling:Stockholm",
  "Christopher Af Bjur:Stockholm",
  "Dan Iresjö:Stockholm",
  "Daniel Albinsson:Lund",
]
`;

exports[`useEmployees > when employees are fetched successfully > should update filter successfully 1`] = `
[
  "Alban Nurkollari:Helsingborg",
  "Anders Ringqvist:Lund",
  "Andreas Håkansson:Helsingborg",
  "Andreas Nilsson:Helsingborg",
  "Andreas Voigt:Helsingborg",
  "Christoffer Hydling:Stockholm",
  "David Tiselius:Helsingborg",
  "Erik Juhlin:Helsingborg",
  "Esma Klay:Helsingborg",
  "Farhad Mehryari:Helsingborg",
  "Filip Johansson:Helsingborg",
  "Henning Gross:Lund",
  "Ingemar Parra:Stockholm",
  "Jimmie Espling:Stockholm",
  "Johan Olsson:Helsingborg",
  "Kanaan Bahmani:Helsingborg",
  "Mario Ingemann:Stockholm",
  "Martin Lekvall:Helsingborg",
  "Maximilian Falkenström:Helsingborg",
  "Mikael Nilsson:Helsingborg",
  "Mohamed Shafiq:Helsingborg",
  "Nick Sandgren:Helsingborg",
  "Niclas Fredriksson:Helsingborg",
  "Niklas Sjöberg:Helsingborg",
  "Oscar Nilsson:Helsingborg",
  "Patrik Palmér:Helsingborg",
  "Peter Heiberg:Helsingborg",
  "Philip Englund:Helsingborg",
  "Remi Tonning:Borlänge",
  "Samir Sliti:Helsingborg",
  "Sofi Persson:Helsingborg",
  "Uday Pratap Singh:Lund",
]
`;

exports[`useEmployees > when employees are fetched successfully > should update sort successfully 1`] = `
[
  "Anders Hedlund:Borlänge",
  "Anders Liljekvist:Borlänge",
  "Edvin Trönnberg:Borlänge",
  "Emil Tylén:Borlänge",
  "Erik Zetterqvist:Borlänge",
  "Jesper Turesson:Borlänge",
  "Johan Karlsson:Borlänge",
  "Jon Bennevall:Borlänge",
  "Klas Broberg:Borlänge",
  "Lewis Clarke:Borlänge",
  "Martin Friman:Borlänge",
  "Mats Törnberg:Borlänge",
  "Mikael Fehrm:Borlänge",
  "Niklas Hedlund:Borlänge",
  "Niklas Stadigs:Borlänge",
  "Remi Tonning:Borlänge",
  "Richard Hartfield:Borlänge",
  "Simon Andersson:Borlänge",
  "Simon Farås:Borlänge",
  "Simon Palmberg:Borlänge",
  "Thomas Brunström:Borlänge",
  "Alban Nurkollari:Helsingborg",
  "Andreas Håkansson:Helsingborg",
  "Andreas Nilsson:Helsingborg",
  "Andreas Voigt:Helsingborg",
  "David Tiselius:Helsingborg",
  "Erik Juhlin:Helsingborg",
  "Esma Klay:Helsingborg",
  "Farhad Mehryari:Helsingborg",
  "Filip Johansson:Helsingborg",
  "Johan Olsson:Helsingborg",
  "Kanaan Bahmani:Helsingborg",
  "Martin Lekvall:Helsingborg",
  "Maximilian Falkenström:Helsingborg",
  "Mikael Nilsson:Helsingborg",
  "Mohamed Shafiq:Helsingborg",
  "Nick Sandgren:Helsingborg",
  "Niclas Fredriksson:Helsingborg",
  "Niklas Sjöberg:Helsingborg",
  "Oscar Nilsson:Helsingborg",
  "Patrik Palmér:Helsingborg",
  "Peter Heiberg:Helsingborg",
  "Philip Englund:Helsingborg",
  "Samir Sliti:Helsingborg",
  "Sofi Persson:Helsingborg",
  "Ani Subotić:Ljubljana",
  "Anže Stražar:Ljubljana",
  "Biserka Cvetkovska:Ljubljana",
  "Darko Lačen:Ljubljana",
  "Denis Ekart:Ljubljana",
]
`;
